### ISSUE TITLE
Not working with pip

### ISSUE THREAD

**Issue by banesullivan** [marcoroth/dependabot-bump-together-action#9]
I can't get more than one of the listed packages to be updated in the produced PR.

For example https://github.com/ResonantGeoData/RD-OpenGeo/pull/42

| PR Comment says multiple are updated | diff says otherwise |
|---|---|
| ![Screen Shot 2022-01-19 at 12 47 23 AM](https://user-images.githubusercontent.com/22067021/150086673-96bacbf7-3585-4091-89f9-938815aa6237.png) | ![Screen Shot 2022-01-19 at 12 47 30 AM](https://user-images.githubusercontent.com/22067021/150086691-f99494b3-2b68-4a80-b0e6-5ffd4855b15a.png) |

from this run: https://github.com/ResonantGeoData/RD-OpenGeo/runs/4864405534?check_suite_focus=true

off of my fork and `master` in this repo

**Comment by marcoroth** [comment]
Hey @banesullivan, thanks for the detailed report!

This is actually a case I somehow haven't considered but should 100% be supported. 

As you see in your first screenshot, it detected all the packages and to what version they should be updated to. I somehow only "added" the last changeset to the actual commit. I pushed up an updated version of the action (it's still building it on the fly, but I think we can address that in a second step).

Can you give this a shot and see if that updates all of the packages you provided? All the arguments are still identical

```yml
      # ...
      - uses: marcoroth/dependabot-bump-together-action@multiple-dependencies-per-file
      # ...
```

Thank you!

**Comment by banesullivan** [comment]
I spent an embarrassing amount of time debugging this to no prevail so I really appreciate you looking into this.

Unfortunately, it seems like those changes are not quite it: https://github.com/ResonantGeoData/RD-OpenGeo/pull/50/files

It switched from being the first dep to the last if that helps debug it. I think the trick is combining the `updated_files_global` rather than taking the unique one as if you look at each of their `content` it's a different version of the same file where the respective dependency was bumped

**Comment by banesullivan** [comment]
I've been looking at the approach in https://github.com/boozallen/dependency-update-action/blob/main/update-script.rb

It looks like `UpdateCheckers` can check more than one dependency at a time.

We could do something like:

https://github.com/boozallen/dependency-update-action/blob/3379c8de85956b811239d802a54b77e47aeefdbd/update-script.rb#L81

To check if the dep matches the given names.

**Comment by banesullivan** [comment]
If it helps, I am running this locally for this repo: https://github.com/banesullivan/dependabot-test

Here is the output:

```
root@a34486c9be89:/action/lib# ruby action.rb
INFO: using user: banesullivan
INFO: using repo: banesullivan/dependabot-test
INFO: using directory: /
INFO: using branch: main
INFO: using package managers: pip
INFO: processing packages: django-rgd, django-rgd-3d, django-rgd-fmv, django-rgd-geometry, django-rgd-imagery
INFO: processing dependency django-rgd-3d 0.2.13
INFO: django-rgd-3d will be updated
INFO: processing dependency django-rgd-fmv 0.2.13
INFO: django-rgd-fmv will be updated
INFO: processing dependency django-rgd-geometry 0.2.13
INFO: django-rgd-geometry will be updated
INFO: processing dependency django-rgd-imagery 0.2.13
INFO: django-rgd-imagery will be updated
INFO: updated pip dependency django-rgd-3d from  (0.2.13) to  (0.2.15) in setup.py
INFO: updated pip dependency django-rgd-fmv from  (0.2.13) to  (0.2.15) in setup.py
INFO: updated pip dependency django-rgd-geometry from  (0.2.13) to  (0.2.15) in setup.py
INFO: updated pip dependency django-rgd-imagery from  (0.2.13) to  (0.2.15) in setup.py
INFO: going to commit changes in setup.py
INFO: going to commit changes in setup.py
INFO: going to commit changes in setup.py
INFO: going to commit changes in setup.py
ERROR: PR already exists or an error has occurred
```

It looks like something goes wrong when making the PR as `pr&.status == 201` is false

**Comment by banesullivan** [comment]
I got it to work!! The trick was having the `FileUpdaters` iterate over the dependencies and sequentially updating the files one after another. See https://github.com/banesullivan/dependabot-test/pull/5/files for a demo

I will shortly open a PR into your `multiple-dependencies-per-file` branch

**Comment by banesullivan** [comment]
@marcoroth, I've somewhat diverged with development and have made my fork production-ready: https://github.com/banesullivan/dependabot-bump-together-action

I hate to diverge the work like this and would be happy to help implement the same changes here if you have the bandwidth to maintain this

**Comment by marcoroth** [comment]
Hey @banesullivan, thank you for all your hard work on this! ❤️ 

I'm going to cross-check if your changes are also working as expected in my projects! If so, I would be super happy to merge all your improvements in!

**Comment by banesullivan** [comment]
> I'm going to cross-check if your changes are also working as expected in my projects! 

Definitely a good idea - I've only tested this against `pip` and still need to test if it'll work for other package managers. I ABSOLUTELY NEED this working with pip... having for npm would also be nice but pip is my top priority

**Comment by marcoroth** [comment]
Okay, I've tested both `bundler` and `npm_and_yarn` and it's looking good 🎉 

Thanks to your changes it now also supports updating multiple packages from the same ecosystem. To this point I just used it to whip up a PR that updates one dependency in `npm_and_yarn` and one in `bundler`.

I also think that we can drop the unnecessary `bundler_version` and `dependabot_version` input arguments and just ship the action from the GitHub container registry.
